codeSnippets/snippets/opentelemetry/server/src/main/kotlin/Monitoring.kt

```diff
@@ -0,0 +1,46 @@
+package com.example
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.opentelemetry.api.trace.SpanKind
+import io.opentelemetry.instrumentation.ktor.v3_0.KtorServerTelemetry
+
+fun Application.configureMonitoring() {
+    val openTelemetry = getOpenTelemetry(serviceName = "opentelemetry-ktor-sample-server")
+    
+    install(KtorServerTelemetry) {
+        setOpenTelemetry(openTelemetry)
+    
+        capturedRequestHeaders(HttpHeaders.UserAgent)
+    
+        spanKindExtractor {
+            if (httpMethod == HttpMethod.Post) {
+                SpanKind.PRODUCER
+            } else {
+                SpanKind.CLIENT
+            }
+        }
+    
+        attributesExtractor {
+            onStart {
+                attributes.put("start-time", System.currentTimeMillis())
+            }
+            onEnd {
+                attributes.put("end-time", System.currentTimeMillis())
+            }
+        }
+    }
+    routing {
+        get("/hello") {
+            call.respondText("Hello World!")
+        }
+        
+        post("/post") {
```

Comment:
I think we need to write server routes: `/known-methods`, `/captured-headers` etc

https://github.com/ktorio/ktor-samples/blob/main/opentelemetry/server/src/main/kotlin/opentelemetry/ktor/example/plugins/routing/Routing.kt